### PR TITLE
Feature/adapter v3

### PR DESCRIPTION
**Pull request by gorbuvla**
Now lib is based on Retrofit 2.6 and does not deal with converting calls to Deferreds. Now its retrofit job to make a suspending fun out of call, we are building a callable chain out of `CallDelegates` now

### CODE REVIEW

File: coroutineadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt [hunk 1]
```diff
@@ -26,12 +25,10 @@ class CallChainImpl(
     override val annotations: Array<out Annotation>
         get() = annotationArray
 
-    override fun proceed(call: Call<*>): Deferred<*> {
-        if (chainIndex + 1 > interceptors.size) {
-            throw IllegalStateException("chainIndex ${chainIndex + 1} does not match with actual interceptor size ${interceptors.size}")
-        }
+    override fun proceed(call: Call<*>): Call<*> {
+        check(chainIndex + 1 > interceptors.size) {"chainIndex ${chainIndex + 1} does not match with actual interceptor size ${interceptors.size}" }
```
Comment (davidbilik) [inline coroutineadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt]: I like the extension `lastIndex`, you dont have to perform any +1 magic.

File: coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt [hunk 2]
```diff
@@ -13,11 +14,13 @@ class OAuthInterceptor internal constructor(private val oAuthStore: OAuthStore)
         val originalRequest = chain.request()
         val builder = originalRequest.newBuilder()
 
+        Log.e(OAuthInterceptor::class.java.simpleName, "Intercepting, token: ${oAuthStore.accessToken}")
```
Comment (davidbilik) [inline coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt]: needed?

File: build.gradle [hunk 3]
```diff
@@ -6,7 +6,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:3.3.2"
+        classpath 'com.android.tools.build:gradle:3.6.0-beta04'
```
Comment (davidbilik) [inline build.gradle]: i would maybe use latest stable here

Comment (gorbuvla) [reply build.gradle]: one who doesnt risk doesnt drink 🍾

File: coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt (hunk 2 above)
Comment (gorbuvla) [reply coroutineoauth/src/main/java/cz/ackee/ackroutine/core/OAuthInterceptor.kt]: ✅

File: coroutineadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt (hunk 1 above)
Comment (gorbuvla) [reply coroutineadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt]: ✅

File: build.gradle (hunk 3 above)
Comment (gorbuvla) [reply build.gradle]: ✅

**Pull request merged**